### PR TITLE
posix: add logging to pthread, pthread_mutex, and pthread_cond

### PR DESCRIPTION
**Pull request by cfriedt**
For A Long Time™️ there was zero logging in our pthread implementation. This made it sometimes difficult to know what was happening under the hood with the POSIX library.

Here, we add error logging to the primary paths of interest in POSIX threads - namely to `pthread_t`, `pthread_mutex_t`, and `pthread_cond_t` operations.

Debug-level logging is also implemented when more verbose output is desired.

Logging is also available by default for any of the pooled resources via the Kconfig template, although it has not yet been implemented for less frequently-traveled types like `pthread_barrier_t`, `pthread_rwlock_t`, and `pthread_spinlock_t`. Those can be implemented on an as-needed basis.

For implementing various APIs on top of pthreads, such as ISO C, and ISO C++ threads, this kind of logging is quite helpful.

### CODE REVIEW

File: lib/posix/pthread.c [hunk 1]
```diff
@@ -662,7 +694,10 @@ int pthread_join(pthread_t pthread, void **status)
 	/* other possibilities? */
 	__ASSERT_NO_MSG(err == 0);
 
+	LOG_DBG("Joined  pthread %x", pthread);
```
Comment (ycsin) [inline lib/posix/pthread.c:697]:
```suggestion
	LOG_DBG("Joined pthread %x", pthread);
```

Comment (cfriedt) [reply lib/posix/pthread.c:697]: That spacing is kind of intentional to get related messages to align vertically. I can remove if necessary though.

File: lib/posix/mutex.c [hunk 2]
```diff
@@ -148,18 +160,24 @@ static int acquire_mutex(pthread_mutex_t *mu, k_timeout_t timeout)
...
 		}
 	}
 
 	if (ret < 0) {
+		LOG_ERR("k_mutex_unlock() failed: %d", ret);
 		ret = -ret;
 	}
 
+	if (ret == 0) {
+		LOG_DBG("Locked mutex  %x", *mu);
```
Comment (ycsin) [inline lib/posix/mutex.c:178]:
```suggestion
		LOG_DBG("Locked mutex %x", *mu);
```

Comment (cfriedt) [reply lib/posix/mutex.c:178]: Same as https://github.com/zephyrproject-rtos/zephyr/pull/62771#discussion_r1328695189

**Review by ycsin — DISMISSED** [review]
just nits

File: lib/posix/mutex.c (hunk 2 above)
Comment (ycsin) [reply lib/posix/mutex.c:178]: ops

**Comment by cfriedt** [comment]
* removed extra spaces from strings (were there for vertical alignment, but I think those are removed anyway)
* added a few debug logs to `cond.c` for more verbose messaging of wait, signal, and broadcast operations

**Comment by cfriedt** [comment]
* log pointer value for actual k_thread, k_mutex, or k_condvar when possible

**Review by ycsin — APPROVED** [review]

**Review by fabiobaltieri — APPROVED** [review]

**Pull request merged**